geode-web-management/src/main/java/org/apache/geode/management/internal/rest/controllers/RootManagementController.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.rest.controllers;
+
+import static org.apache.geode.management.api.RestfulEndpoint.URI_CONTEXT;
+import static org.apache.geode.management.api.RestfulEndpoint.URI_VERSION;
+import static org.apache.geode.management.internal.Links.API_ROOT;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Optional;
+
+import io.swagger.annotations.ApiOperation;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping;
+
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.management.api.ClusterManagementResult;
+import org.apache.geode.management.internal.Links;
+
+@RestController("api-root")
+@RequestMapping(URI_VERSION)
+public class RootManagementController extends AbstractManagementController {
```

Comment:
In order to not complicate this PR, let's rip out all the root links first.

Reply:
when you say "rip out" do you mean split into two PRs, or do you mean get rid of forever?

Reply:
I mean if we don't have this root link requirement, the implementation might be easier. But if you didn't think it would make things easier, you can leave them here.

Reply:
PR was split, so this PR is now *just* the root controller